addons/website/controllers/model_page.py

```diff
@@ -0,0 +1,134 @@
+import ast
+
+import werkzeug
+
+from odoo.addons.http_routing.models.ir_http import slug, unslug
+from odoo.http import Controller, request, route
+from odoo.osv.expression import AND, OR
+
+
+class ModelPageController(Controller):
+    pager_step = 20
+
+    @route([
+        "/model/<string:page_name_slugified>",
+        "/model/<string:page_name_slugified>/page/<int:page_number>",
+        "/model/<string:page_name_slugified>/<string:record_slug>",
+    ], website=True, auth="public")
+    def generic_model(self, page_name_slugified=None, page_number=1, record_slug=None, **searches):
+        if not page_name_slugified:
+            raise werkzeug.exceptions.NotFound()
+
+        website = request.website
+
+        page_type = "listing"
+        if record_slug is not None:
+            page_type = "single"
+
+        website_page_domain = AND([
+            [("page_type", "=", page_type)],
+            [("name_slugified", "=", page_name_slugified)],
+            [("website_published", "=", True)],
+            website.website_domain(),
+        ])
+
+        page = request.env["website.controller.page"].sudo().search(website_page_domain, limit=1)
+        if not page:
+            raise werkzeug.exceptions.NotFound()
+
+        view = page.view_id
+        if not view:
+            raise werkzeug.exceptions.NotFound()
+
+        target_model_name = page.model_id.model
+        Model = request.env[target_model_name]
+        if not Model.check_access_rights("read", raise_exception=False):
+            raise werkzeug.exceptions.NotFound()
```

Comment:
Don't you want to show a forbidden page instead?

Reply:
This is a detail. I took example on github, where 404 are raised when it genuinely doesn't exist or when we don't have have rights.

Reply:
Everything is a detail, it's just a software at the end :wink: 
All details matter tho. But yeah, feel free to ignore it.

> I took example on github

Cool, but what about taking exemple on Odoo where records raise a 403 when you don't have access but it exists?

Reply:
Well if it's canon, I don't mind of course.... Thing is, with Odoo, canonification very rarely happens